plugin/metrics/metrics.go

```diff
@@ -104,6 +104,18 @@ func (m *Metrics) OnStartup() error {
 	return nil
 }
 
+// OnRestart stops the listener on reload.
+func (m *Metrics) OnRestart() error {
+	// relingish our listener as we re-listen on successfull reload
+	if m.ln != nil {
+		if err := m.ln.Close(); err != nil {
+			return err
+		}
+		m.ln = nil
+	}
+	return nil
+}
+
```

Comment:
So.. the OnRestart is the same (a little proper) than the OnShutdown .. no ? Maybe we can have only one to be called for the 2 events:
- onRestart
- onFinalShutdown
I would not report the error to Caddy as it will prevent to continue the RESTART or SHUTDOWN. Anyway if the listener cannot be closed it means it is already closed. Maybe INFO the  error.

NOTE: onShutdown works too .. but is confusing for the reader.